zwave_js_server/model/controller/__init__.py

```diff
@@ -420,13 +420,16 @@ async def async_is_failed_node(self, node_id: int) -> bool:
         return cast(bool, data["failed"])
 
     async def async_get_association_groups(
-        self, node_id: int
+        self, source: Association
     ) -> Dict[int, AssociationGroup]:
         """Send getAssociationGroups command to Controller."""
         data = await self.client.async_send_command(
             {
                 "command": "controller.get_association_groups",
-                "nodeId": node_id,
+                "nodeId": source.node_id,
+                **(
```

Comment:
this syntax is a little hard to read. Instead of conditionally adding the key this way, can we create a payload dict and then:
```
if source.endpoint is not None:
  payload["endpoint"] = source.endpoint
```
instead?